control-plane-operator/controllers/hostedcontrolplane/v2/kas/deployment.go

```diff
@@ -0,0 +1,465 @@
+			"--tls-key=/var/run/app/certs/tls.key",
+			"--token-audience=openshift",
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("10m"),
+				corev1.ResourceMemory: resource.MustParse("25Mi"),
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: awsPodIdentityWebhookServingCertVolumeName, MountPath: "/var/run/app/certs"},
+			{Name: awsPodIdentityWebhookKubeconfigVolumeName, MountPath: "/var/run/app/kubeconfig"},
+		},
+	})
+
+	podSpec.Volumes = append(podSpec.Volumes,
+		corev1.Volume{
+			Name: awsPodIdentityWebhookServingCertVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: manifests.AWSPodIdentityWebhookServingCert("").Name},
+			}},
+		corev1.Volume{
+			Name: awsPodIdentityWebhookKubeconfigVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: manifests.AWSPodIdentityWebhookKubeconfig("").Name},
+			}},
+	)
+}
+
+func buildKASContainerWaitForEtcd(image string, namespace string) corev1.Container {
+	c := corev1.Container{
+		Name: "wait-for-etcd",
+	}
+	c.Image = image
+	c.TerminationMessagePolicy = corev1.TerminationMessageReadFile
+	c.TerminationMessagePath = corev1.TerminationMessagePathDefault
+	c.ImagePullPolicy = corev1.PullIfNotPresent
+	c.Command = []string{
+		"/bin/bash",
+	}
+	c.Args = []string{
+		"-c",
+		waitForEtcdScript(namespace),
+	}
+	return c
+}
+
+func buildAuditLogsContainer() corev1.Container {
+	return corev1.Container{
+		Name:            "audit-logs",
+		Image:           "cli",
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command:         []string{"/bin/bash"},
+		Args: []string{
+			"-c",
+			RenderAuditLogScript("/var/log/kube-apiserver/audit.log"),
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("5m"),
+				corev1.ResourceMemory: resource.MustParse("10Mi"),
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{{
+			Name:      workLogsVolumeName,
+			MountPath: "/var/log/kube-apiserver",
+		}},
+	}
+}
+
+func buildKASCloudConfigVolume(configMapName, providerName string) corev1.Volume {
+	v := corev1.Volume{
+		Name: cloudConfigVolumeName,
+	}
+	if providerName == azure.Provider {
+		v.Secret = &corev1.SecretVolumeSource{SecretName: configMapName}
+	} else {
+		v.ConfigMap = &corev1.ConfigMapVolumeSource{
+			LocalObjectReference: corev1.LocalObjectReference{Name: configMapName},
+			DefaultMode:          ptr.To[int32](420),
+		}
+	}
+	return v
+}
+
+func buildKASAuditWebhookConfigFileVolume(auditWebhookRef *corev1.LocalObjectReference) corev1.Volume {
+	v := corev1.Volume{
+		Name: auditWebhookConfigFileVolumeName,
+	}
+	v.Secret = &corev1.SecretVolumeSource{}
+	v.Secret.SecretName = auditWebhookRef.Name
+	return v
+}
+
+func invokeBootstrapRenderScript(workDir, payloadVersion, featureGateYaml string) string {
+
+	var script = `#!/bin/sh
```

Comment:
It'd be nice if we could make this a static script in the deployment.yaml 
One possibility could be to replace params with env vars

Reply:
done